src/profile/jwks.ts

```diff
@@ -0,0 +1,185 @@
+/**
+ * Copyright 2021 Inrupt Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import { security } from "../constants";
+import {
+  Iri,
+  IriString,
+  Jwk,
+  Jwks,
+  SolidDataset,
+  ThingPersisted,
+  Url,
+  UrlString,
+  WebId,
+  WithResourceInfo,
+} from "../interfaces";
+import { overwriteFile } from "../resource/file";
+import {
+  getSourceUrl,
+  internal_defaultFetchOptions,
+} from "../resource/resource";
+import { getSolidDataset } from "../resource/solidDataset";
+import { getUrl } from "../thing/get";
+import { setIri } from "../thing/set";
+import { getThing, setThing } from "../thing/thing";
+
+function getProfileFromProfileDoc(
+  profileDataset: SolidDataset,
+  webId: WebId
+): ThingPersisted {
+  const profile = getThing(profileDataset, webId);
+  if (profile === null) {
+    throw new Error(
+      `Profile document [${getSourceUrl(
+        profileDataset
+      )}] does not include WebID [${webId}]`
+    );
+  }
+  return profile;
+}
+
+/**
+ * Set a JWKS IRI associated to a WebID in a profile document.
```

Comment:
Maybe do a search and replace in this codebase and the SCA codebase for "associated to", and change to "associated with":
```suggestion
 * Set a JWKS IRI associated with a WebID in a profile document.
```